framework/src/testing/packet.rs

```diff
@@ -0,0 +1,253 @@
+//! proptest strategies to generate packets
+
+use crate::packets::ip::v6::Ipv6;
+use crate::packets::ip::{ProtocolNumber, ProtocolNumbers};
+use crate::packets::{EtherType, EtherTypes, Ethernet, MacAddr, Packet, RawPacket, Tcp};
+use proptest::arbitrary::{any, Arbitrary};
+use proptest::strategy::{Just, Strategy};
+use std::any::Any;
+use std::collections::HashMap;
+use std::net::Ipv6Addr;
+
+/// Enumeration of settable packet fields
+#[allow(non_camel_case_types)]
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub enum field {
```

Comment:
should we include ndps in this?

Reply:
eventually yeah, that's the idea. right now I'm skipping that because the piece I'm working on doesn't need ndp yet.